cpp/tests/test_ide_secir.cpp

```diff
@@ -242,15 +242,181 @@ TEST(IdeSecir, checkSimulationFunctions)
+    // --- Case with forceofinfection.
+    mio::TimeSeries<ScalarType> init_copy(init);
+    mio::isecir::Model model(std::move(init_copy), N, deaths, 0, std::move(parameters));
+    model.check_constraints(dt);
+
+    // Carry out simulation.
+    mio::isecir::Simulation sim(model, 0, dt);
+    sim.advance(tmax);
+
+    mio::TimeSeries<ScalarType> result = sim.get_result();
+
+    // Compare with previous run.
+    for (Eigen::Index i = 0; i < (Eigen::Index)mio::isecir::InfectionState::Count; i++) {
+        EXPECT_NEAR(result[0][i], vec_forceofinfection[i], 1e-8);
+    }
+
+    // --- Case with total_confirmed_cases.
+    mio::TimeSeries<ScalarType> init_copy2(init);
+    mio::isecir::Model model2(std::move(init_copy2), N, deaths, 1000, std::move(parameters));
+    model2.check_constraints(dt);
+
+    // Carry out simulation.
+    mio::isecir::Simulation sim2(model2, 0, dt);
+    sim2.advance(tmax);
+    result = sim2.get_result();
+
+    // Compare with previous run.
+    for (Eigen::Index i = 0; i < (Eigen::Index)mio::isecir::InfectionState::Count; i++) {
+        EXPECT_NEAR(result[0][i], vec_total_confirmed[i], 1e-8);
+    }
+
+    // --- Case with S.
+    contact_matrix[0]                              = mio::ContactMatrix(Eigen::MatrixXd::Constant(1, 1, 0));
+    parameters.get<mio::isecir::ContactPatterns>() = mio::UncertainContactMatrix(contact_matrix);
+
+    mio::TimeSeries<ScalarType> init_copy3(init);
+    mio::isecir::Model model3(std::move(init_copy3), N, deaths, 0, std::move(parameters));
+
+    model3.m_populations.get_last_value()[(Eigen::Index)mio::isecir::InfectionState::Susceptible] = 5000;
+    model3.m_populations.get_last_value()[(Eigen::Index)mio::isecir::InfectionState::Recovered]   = 0;
+
+    model3.check_constraints(dt);
+
+    // Carry out simulation.
+    mio::isecir::Simulation sim3(model3, 0, dt);
+    sim3.advance(tmax);
+    result = sim3.get_result();
+
+    // Compare with previous run.
+    for (Eigen::Index i = 0; i < (Eigen::Index)mio::isecir::InfectionState::Count; i++) {
+        EXPECT_NEAR(result[0][i], vec_S[i], 1e-8);
+    }
+
+    // --- Case with R.
+    mio::TimeSeries<ScalarType> init_copy4(init);
+    mio::isecir::Model model4(std::move(init_copy4), N, deaths, 0, std::move(parameters));
+
+    model4.m_populations.get_last_value()[(Eigen::Index)mio::isecir::InfectionState::Susceptible] = 0;
+    model4.m_populations.get_last_value()[(Eigen::Index)mio::isecir::InfectionState::Recovered]   = 1000;
+
+    model4.check_constraints(dt);
+
+    // Carry out simulation.
+    mio::isecir::Simulation sim4(model4, 0, dt);
+    sim4.advance(tmax);
+    result = sim4.get_result();
+
+    // Compare with previous run.
+    for (Eigen::Index i = 0; i < (Eigen::Index)mio::isecir::InfectionState::Count; i++) {
+        EXPECT_NEAR(result[0][i], vec_R[i], 1e-8);
+    }
+
+    // --- Case without fitting initialization method.
+    mio::isecir::Model model5(std::move(init), N, deaths, 0, std::move(parameters));
```

Comment:
I would make this test more explicit and set init to the zero vector. If you use move twice you don't know what you are putting in your model. In theory it could be possible that this is a valid init vector for the forceofinfection method.

Reply:
the contact rate was previously set to 0. Therefore the vector can be valid here. If the force infection method worked, it would also be used in cases S and R. Should I still make this more explicit and pass the zero vector? 
That's the first and last time i use move with the vector, so this is ok, isn't it?

Reply:
What do you mean with the sentence "If the force infection method worked, it would also be used in cases S and R."?

You're right, if the contact rate is set to 0 it doesn't matter what vector is used for init because the forceofinfection will always be 0. Then it's fine if you use move, I think a comment would still be helpful here and make it more obvious :)

Reply:
> What do you mean with the sentence "If the force infection method worked, it would also be used in cases S and R."?
>

If the forceofinfection was greater than zero, this method would also be used to initialize in the tests for the initialization with S and R (see lines 353-395).  I think the added comment in line 354f. makes this more clear :) 

> You're right, if the contact rate is set to 0 it doesn't matter what vector is used for init because the forceofinfection will always be 0. Then it's fine if you use move, I think a comment would still be helpful here and make it more obvious :)

Added a comment.